jetty-server/src/main/java/org/eclipse/jetty/server/session/FileSessionDataStore.java

```diff
@@ -99,117 +115,162 @@ public void setDeleteUnrestorableFiles(boolean deleteUnrestorableFiles)
-                if (!StringUtil.isBlank(idWithContext))
-                    idsWithContext.add(idWithContext);
-                return true;
-            }
-        });
-        
-       
-        //got the list of all sessionids with their contexts, remove all old files for each one
-        for (String idWithContext:idsWithContext)
+        //iterate over the files and work out which have expired
+        for (String filename:_sessionFileMap.values())
         {
-            deleteOldFiles(_storeDir, idWithContext);
-        }
-
-        
-        //now find sessions that have expired in any context
-        files = _storeDir.listFiles(new FilenameFilter() 
-        {      
-            @Override
-            public boolean accept(File dir, String name)
+            try
             {
-                if (dir != _storeDir)
-                    return false;
-
-                //dir may contain files that don't match our naming pattern
-                if (!match(name))
-                    return false;
-                
-                try
-                {
-                    long expiry = getExpiryFromString(name);
-                    return expiry > 0 && expiry < now;
-                }
-                catch (Exception e)
-                {
-                    return false;
-                }
+                long expiry = getExpiryFromFilename(filename);
+                if (expiry > 0 && expiry < now)
+                    expired.add(getIdFromFilename(filename));
             }
-        });
-        
-        if (files != null)
-        {
-            for (File f:files)
+            catch (Exception e)
             {
-                expired.add(getIdFromFile(f));
+                LOG.warn(e);
             }
         }
         
-        //check candidates that were not found to be expired, perhaps they no
-        //longer exist and they should be expired
+        //check candidates that were not found to be expired, perhaps 
+        //because they no longer exist and they should be expired
         for (String c:candidates)
         {
             if (!expired.contains(c))
             {
-                //check if the file exists
-                File f = getFile(_storeDir, c);
-                if (f == null || !f.exists())
+                //if it doesn't have a file then the session doesn't exist
+                String filename = _sessionFileMap.get(getIdWithContext(c));
+                if (filename == null)
                     expired.add(c);
             }
         }
-        
+
+        //Infrequently iterate over all files in the store, and delete those
+        //that expired a long time ago, even if they belong to
+        //another context. This ensures that files that
+        //belong to defunct contexts are cleaned up. 
+        //If the graceperiod is disabled, don't do the sweep!
+        if ((_gracePeriodSec > 0) && ((_lastSweepTime == 0) || ((now - _lastSweepTime) >= (5000L*_gracePeriodSec))))
```

Comment:
rather than `5000L*_gracePeriodSec`, I would prefer to see `5*TimeUnit.SECONDS.toMillis(_gracePeriodSec)`

Reply:
OK, done.